src/renderers/common/RenderObject.js

```diff
@@ -233,10 +233,22 @@ export default class RenderObject {
 
 		}
 
-		const itemCount = hasIndex === true ? index.count : geometry.attributes.position.count;
+		const position = geometry.attributes.position;
+
+		if ( hasIndex ) {
+
+			const itemCount = index.count;
+			firstVertex = Math.max( firstVertex, 0 );
+			lastVertex = Math.min( lastVertex, itemCount );
+
+		} else if ( position !== undefined && position !== null ) {
+
+			const itemCount = position.count;
+			firstVertex = Math.max( firstVertex, 0 );
+			lastVertex = Math.min( lastVertex, itemCount );
+
+		}
```

Comment:
Could this be written like so?
```js
const position = geometry.attributes.position;
let itemCount;

if ( hasIndex ) {

	itemCount = index.count;

} else if ( position !== undefined && position !== null ) {

	itemCount = position.count;

}

firstVertex = Math.max( firstVertex, 0 );
lastVertex = Math.min( lastVertex, itemCount );
```
If so, what is the value of `itemCount` when no index and position attribute are present?

Reply:
Even better! Done.